common/views/components/ArchiveTree/ArchiveTree.js

```diff
@@ -249,44 +278,34 @@ type ListItemType = {|
   isRootItem: boolean,
 |};
 
-// temp component to show buttons when appropriate until API changes made
-const ButtonContainer = ({
-  item,
-  toggles,
-  setCollectionTree,
-  fullTree,
-  setShowButton,
-  children,
-}) => {
-  useEffect(() => {
-    expandTree(
-      item.work.id,
-      toggles,
-      setCollectionTree,
-      fullTree,
-      setShowButton
-    );
-  }, []);
-  return children;
-};
 const ListItem = ({
   item,
   setShowArchiveTreeModal,
   currentWorkId,
   selected,
   setCollectionTree,
   fullTree,
-  collectionTree,
   isRootItem,
 }: ListItemType) => {
-  const [showNested, setShowNested] = useState(item.openByDefault || false);
   const [showButton, setShowButton] = useState(item.children);
+  const toggles = useContext(TogglesContext);
+  useEffect(() => {
+    let isMounted = true;
```

Comment:
What is this doing? Shouldn't the whole `checkForChildren` be wrapped in this? It _looks_ like we're trying to only do this on initial load, which having the deps of the `useState` set to `[]` should do.